src/cpu/cgroup.rs

```diff
@@ -76,129 +76,45 @@ pub fn read() -> Result<CgroupCpuMeasurement> {
     os::read()
 }
 
-#[cfg(target_os = "linux")]
+// #[cfg(target_os = "linux")]
 mod os {
-    use super::super::super::{
-        dir_exists, file_to_buf_reader, parse_u64, path_to_string, precise_time_ns,
-        read_file_value_as_u64, Result,
-    };
-    use super::{CgroupCpuMeasurement, CgroupCpuStat};
+    use super::super::cgroup_v1::os::read_and_parse_v1_sys_stat;
+    use super::super::cgroup_v2::os::read_and_parse_v2_sys_stat;
+    use super::CgroupCpuMeasurement;
     use crate::error::ProbeError;
-    use std::io::BufRead;
+    use crate::{dir_exists, Result};
     use std::path::Path;
 
-    const CPU_SYS_NUMBER_OF_FIELDS: usize = 2;
-
     pub fn read() -> Result<CgroupCpuMeasurement> {
-        let sys_fs_dir = Path::new("/sys/fs/cgroup/cpuacct/");
-        if dir_exists(sys_fs_dir) {
-            read_and_parse_sys_stat(&sys_fs_dir)
-        } else {
-            let message = format!(
-                "Directory `{}` not found",
-                sys_fs_dir.to_str().unwrap_or("unknown path")
-            );
-            Err(ProbeError::UnexpectedContent(message))
+        let v1_sys_fs_dir = Path::new("/sys/fs/cgroup/cpuacct/");
+        if dir_exists(v1_sys_fs_dir) {
+            return read_and_parse_v1_sys_stat(&v1_sys_fs_dir);
```

Comment:
With v2 becoming the new thing and default. Switching this around would save us a check for the old v1 files. Move the order around of the v1 and v2 checks.